src/main/java/org/cbioportal/application/security/config/Saml2SecurityConfig.java

```diff
@@ -89,12 +92,24 @@ public OpenSaml4AuthenticationProvider openSaml4AuthenticationProvider() {
       Saml2Authentication authentication = delegate.convert(responseToken);
       var principal =
           (Saml2AuthenticatedPrincipal) Objects.requireNonNull(authentication).getPrincipal();
-      Collection<String> roles = principal.getAttribute(this.roleAttributeName);
+
+      String username = (String) principal.getAttribute("username").getFirst();
+
+      User cbioUser = securityRepository.getPortalUser(username);
+
+      if (cbioUser == null) {
+        Saml2Authentication sm2fail =
+            new Saml2Authentication(principal, authentication.getSaml2Response(), new HashSet<>());
+        sm2fail.setAuthenticated(false);
+        return sm2fail;
+      }
+
+      UserAuthorities authorities = securityRepository.getPortalUserAuthorities(username);
+
       Set<GrantedAuthority> mappedAuthorities = new HashSet<>();
-      if (!Objects.isNull(roles)) {
-        mappedAuthorities.addAll(GrantedAuthorityUtil.generateGrantedAuthoritiesFromRoles(roles));
-      } else {
-        mappedAuthorities.addAll(authentication.getAuthorities());
+
+      if (!Objects.isNull(authorities)) {
+        mappedAuthorities.addAll(AuthorityUtils.createAuthorityList(authorities.getAuthorities()));
       }
```

Comment:
I don't quite understand this change, are we going to read authories from database only? I think roles we get from `Collection<String> roles = principal.getAttribute(this.roleAttributeName);` is also important, right?